### ISSUE TITLE
`link rel=stylesheet` inside a `noscript` element causes `adjacentNodeOrRelation` to be undefined

### ISSUE THREAD

**Issue by philwolstenholme** [assetgraph/assetgraph#1251]
I'm using assetgraph `7.10.0`, via Subfont `6.12.1`, via the Subfont Netlify plugin.

I have a Vite plugin that wraps some a CSS `link` element in a `noscript` element as part of a no-JS fallback for some async loading of non-critical CSS. It is appended to the very end of my document:

```html
<noscript><link rel=stylesheet href=/assets/global.ebed6fb3.css></noscript></body></html>
```

If I swap the Vite plugin to inject this markup instead then the issue goes away:

```html
<link rel=stylesheet href=/assets/global.ecb97292.css></body></html>
```

When the `noscript` element is added back then the error returns:

```
TypeError: Cannot read properties of undefined (reading 'isRelation')

In "onPostBuild" event in "netlify-plugin-subfont" from netlify.toml and package.json
      at HtmlPreloadLink.attach (/Users/phil/Sites/wolstenhol-11ty/node_modules/assetgraph/lib/relations/HtmlRelation.js:134:36)
      at HtmlPreloadLink.attach (/Users/phil/Sites/wolstenhol-11ty/node_modules/assetgraph/lib/relations/HtmlResourceHint.js:186:11)
      at HtmlPreloadLink.attach (/Users/phil/Sites/wolstenhol-11ty/node_modules/assetgraph/lib/relations/HtmlPreloadLink.js:25:11)
      at Html.addRelation (/Users/phil/Sites/wolstenhol-11ty/node_modules/assetgraph/lib/assets/Asset.js:1115:16)
      at subsetFonts (/Users/phil/Sites/wolstenhol-11ty/node_modules/subfont/lib/subsetFonts.js:1436:26)
      at async subfont (/Users/phil/Sites/wolstenhol-11ty/node_modules/subfont/lib/subfont.js:222:24)
      at async onPostBuild (/Users/phil/Sites/wolstenhol-11ty/node_modules/netlify-plugin-subfont/lib/index.js:69:5)
      at async Object.run (file:///Users/phil/.nvm/versions/node/v16.13.0/lib/node_modules/netlify-cli/node_modules/@netlify/build/lib/plugins/child/run.js:14:5)
      at async handleEvent (file:///Users/phil/.nvm/versions/node/v16.13.0/lib/node_modules/netlify-cli/node_modules/@netlify/build/lib/plugins/child/main.js:28:42)
      at async process.<anonymous> (file:///Users/phil/.nvm/versions/node/v16.13.0/lib/node_modules/netlify-cli/node_modules/@netlify/build/lib/plugins/ipc.js:91:24)
```

**Comment by philwolstenholme** [comment]
This will be hard to reproduce as a bunch of environment variables are needed for the build to get as far as Subfont's step, but the repo is public at least: https://github.com/philwolstenholme/wolstenhol-11ty

Let me know how I can help, I can stick in some console.logs if you provide a patch or line numbers and file names :)

**Comment by papandreou** [comment]
Thanks for a great bug report! This should be fixed in subfont 6.12.2.
